src/internal/clusterstate/v2.5.0/collection.go

```diff
@@ -267,32 +266,84 @@ func migratePostgreSQLCollection(ctx context.Context, tx *pachsql.Tx, name strin
 		vals[oldKey] = pair{newKey, proto.Clone(newVal)}
 
 	}
-	log.Info(ctx, fmt.Sprintf("Migrating collection.%s", name))
-	i := 0
+
+	// Optimized code for building the UPDATEs and sending them to Postgres in batches
+	// NOTE: This code necessitated upgrading from sqlx 1.2.0 to sqlx 1.3.x to get batching support for named variables
+
+	log.Info(ctx, fmt.Sprintf("Migrating %s [%d rows]", name, len(vals)))
+
+	batch_size := 1000 // Maximum size of each batch
```

Comment:
[Variables should be named with `lowerCamelCase`](https://go.dev/doc/effective_go#mixed-caps).

Reply:
Old programming habits die hard.......